dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/constraints/DatadogDataConstraints.kt

```diff
@@ -32,32 +32,49 @@ internal class DatadogLogConstraints : LogConstraints {
         return convertedTags.take(MAX_TAG_COUNT)
     }
 
-    override fun validateAttributes(attributes: Map<String, Any?>): Map<String, Any?> {
+    override fun validateAttributes(
+        attributes: Map<String, Any?>,
+        maxKeyDepthLevel: Int,
+        attributesGroupName: String?
+    ): Map<String, Any?> {
         val convertedAttributes = attributes.mapNotNull {
-            val key = convertAttributeKey(it.key)
-            if (key == null) {
+            if (it.key == null) {
                 devLogger.e("\"$it\" is an invalid attribute, and was ignored.")
                 null
-            } else {
-                if (key != it.key) {
-                    devLogger.w(
-                        "attribute \"${it.key}\" " +
-                            "was modified to \"$key\" to match our constraints."
-                    )
-                }
-                key to it.value
             }
+            val key = convertAttributeKey(it.key, maxKeyDepthLevel)
+            if (key != it.key) {
+                devLogger.w(
+                    "Key \"${it.key}\" " +
+                        "was modified to \"$key\" to match our constraints."
```

Comment:
in case of `extraInfo`, this line will print seemingly valid attributes (only depth of 8) by saying `was modified to match our constraints`

i think we should log the actual attribute key here
example: instead of `extra-info`, we should log `usr.extra-info` because that's what we send to intake

Reply:
User doesn't know about the prefix he only knows about the key names his providing so I don't see the point in adding this prefix.

Reply:
On the other hand I see your point ...if we decide to give a warning with the whole key I will remove this complexity and only check for max depth = 10 for all the attributes.
@ncreated @xgouchet  What do you think ?

Reply:
Nevermind what I said before I can remove the complexity and come back with a different approach and also keep the relevant warning message ;)